2022.9.30
学习C++primer  5.5跳转语句  5.6 try语句块和异常处理

## 跳转语句
C++提供4中跳转语句：break, continue, goto, return。

goto语句的作用是从goto语句无条件跳转到同一函数内的另一条语句。

```c++
goto label;
label :
```

向后跳过一个已经执行的定义是合法的。跳回到变量之前意味着系统将销毁该变量，然后重新创建它。

## try语句块和异常处理
异常是指存在于运行时的反常行为，这些行为超出了函数正常功能的范围。

典型的异常包括失去数据库连接以及遇到意外输入等。

当程序的某部分检测到一个它无法处理的问题时，需要用到异常处理。此时检测出来问题的部分应该发出某种信号以表明程序遇到了故障，无法继续下去了，而且信号的发出方无需知道故障将在何处得到解决。

如果程序中含有可能引发异常的代码，那么通常也会有专门的代码处理问题。例如，如果程序的问题是输入无效，则异常处理部分可能会要求用户重新输入正确的数据；如果丢失了数据库连接，会发出报警信息。

异常处理机制为程序中异常检测和异常处理这两部分的协作提供支持。

在C++语言中，异常处理包括：

throw表达式，异常检测部分使用throw表达式来表示它遇到了无法处理的问题。

try语句块，异常处理部分使用try语句块处理异常。try语句块中代码抛出的异常通常会被某个catch子句处理。

一套异常类，用于在throw表达式和相关的catch子句之间传递异常的具体信息。

